src/Ingester/Ingester/LtiInstanceIngester.php

```diff
@@ -23,21 +23,23 @@
 namespace OAT\SimpleRoster\Ingester\Ingester;
 
 use OAT\SimpleRoster\Entity\EntityInterface;
-use OAT\SimpleRoster\Entity\Infrastructure;
+use OAT\SimpleRoster\Entity\LtiInstance;
 
-class InfrastructureIngester extends AbstractIngester
+class LtiInstanceIngester extends AbstractIngester
 {
     public function getRegistryItemName(): string
     {
-        return 'infrastructure';
+        return 'lti-instance';
     }
 
     protected function createEntity(array $data): EntityInterface
     {
-        return (new Infrastructure())
-            ->setLabel($data['label'])
-            ->setLtiDirectorLink($data['ltiDirectorLink'])
-            ->setLtiKey($data['ltiKey'])
-            ->setLtiSecret($data['ltiSecret']);
+        return new LtiInstance(
+            0,
```

Comment:
Next stupid question: what would occur in case of ingesting multiple line items with same id (`0`), since entity XML says it should be autogenerated value?

Reply:
The intention is to define entities as immutable objects, but as far as I am concerned there is no other way of doing this. Doctrine is smart enough to know if id is 0, it's a new, not persisted entity. It looks weird here I admit, but the benefit is to not have nullable `getId()` methods (as you pointed out in one of your comments above) + having immutable entity. What do you think?